### PR TITLE
catgirl: new port

### PR DESCRIPTION
**Pull request by ryanakca**
#### Description

new port:

```
description         a TLS-only terminal IRC client

long_description    catgirl is a TLS-only terminal IRC client. Its features \
                    include tab completion, split scrolling, URL detection, nick \
                    colouring topic diffing, and message filtering.
```

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 11.4 20F71 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/catgirl/Portfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+long_description    catgirl is a TLS-only terminal IRC client. Its features \
+                    include tab completion, split scrolling, URL detection, nick \
+                    colouring topic diffing, and message filtering.
+
+homepage            https://git.causal.agency/catgirl/
+master_sites        https://git.causal.agency/catgirl/snapshot/
+
+checksums           rmd160  a2c1da6d88e9cfd4ef039a0b68b991b6100fed05 \
+                    sha256  2ef69606640f25ca695bc65d0dd9e16f5fa36184feb38d1253a5648dad68776b
+
```
Comment (herbygillot) [inline net/catgirl/Portfile:24]: `checksums` should also have a `size` checksum, as called out by `port lint`:

```
port lint --nitpick catgirl:
Warning: missing recommended checksum type: size
```

**Review by herbygillot — CHANGES_REQUESTED** [review]

**Review by herbygillot — APPROVED** [review]
Only other nitpick if any is that maybe `master_sites` could look like the following:

```
homepage            https://git.causal.agency/catgirl/
master_sites        ${homepage}snapshot/
```

...but otherwise looks good.

**Comment by ryanakca** [comment]
Done, thanks for the feedback!

**Comment by herbygillot** [comment]
Thanks for this submission 👍🏾

**Comment by akierig** [comment]
since this port requires LibreSSL instead of OpenSSL, has there been any consideration of having the catgirl author's [LibreTLS](https://git.causal.agency/libretls/) (port of LibreSSL's libtls) library added to the port tree? As an alternative to requiring LibreTLS, I mean.

**Comment by causal-agent** [comment]
@akierig I'm on it

**Pull request merged**